apps/events/models.py

```diff
@@ -0,0 +1,165 @@
+from datetime import timedelta
+
+from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.db.models import F, Func, OuterRef, Q, Subquery
+from django.utils import timezone
+
+from apps.chat.models import ChatMessage, ChatMessageType
+from apps.events.actions import end_conversation, log, summarize_conversation
+from apps.experiments.models import Experiment, ExperimentSession
+from apps.utils.models import BaseModel
+
+ACTION_FUNCTIONS = {"log": log, "end_conversation": end_conversation, "summarize": summarize_conversation}
+
+
+class EventActionType(models.TextChoices):
+    LOG = "log"  # Prints the last message
+    END_CONVERSATION = "end_conversation"  # Ends the conversation
+    SUMMARIZE = "summarize"  # Summarize the conversation
+
+
+class EventAction(BaseModel):
+    action_type = models.CharField(choices=EventActionType.choices)
+    params = models.JSONField(blank=True, default=dict)
+
+
+class EventLogStatusChoices(models.TextChoices):
+    SUCCESS = "success"
+    FAILURE = "failure"
+
+
+class EventLog(BaseModel):
+    # StaticTrigger or TimeoutTrigger
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    object_id = models.PositiveBigIntegerField()
+    content_object = GenericForeignKey("content_type", "object_id")
+
+    session = models.ForeignKey(ExperimentSession, on_delete=models.CASCADE, related_name="event_logs")
+    chat_message = models.ForeignKey(
+        ChatMessage, on_delete=models.CASCADE, related_name="event_logs", null=True, blank=True
+    )
+    status = models.CharField(choices=EventLogStatusChoices.choices)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["content_type", "object_id"]),
+        ]
+
+
+class StaticTriggerType(models.TextChoices):
+    CONVERSATION_END = "conversation_end"
+    LAST_TIMEOUT = "last_timeout"
```

Comment:
Could also add `safety_layer_triggered`, `conversation_start`, `new_participant` (maybe `human_message`, `bot_message`)

Reply:
Totally. Would you like me to add these here already?